From 2892285a2b58570e79ea74d552037482e7751bf4 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Wed, 6 Jun 2018 15:57:05 +0300
Subject: [PATCH 3921/5242] MLK-18516: ARM64: dts: imx8mm: add audio tdm dts

commit  652e0477fb4c29da56c33c449ba2c5aa94c2409f from
https://source.codeaurora.org/external/imx/linux-imx.git

Add a distinct DTS in order to showcase AK4458/AK5558 TDM mode.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 ++-
 .../dts/freescale/fsl-imx8mm-evk-audio-tdm.dts     |   25 ++++++++++++++++++++
 2 files changed, 27 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-audio-tdm.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index d1cac86..53903ce 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -79,4 +79,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
 				 fsl-imx8mm-evk-ak4497.dtb \
 				 fsl-imx8mm-evk-m4.dtb \
-				 fsl-imx8mm-evk-ak5558.dtb
+				 fsl-imx8mm-evk-ak5558.dtb \
+				 fsl-imx8mm-evk-audio-tdm.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-audio-tdm.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-audio-tdm.dts
new file mode 100644
index 0000000..fc1826bb
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-audio-tdm.dts
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mm-evk-ak5558.dts"
+
+/ {
+	sound-ak4458 {
+		fsl,tdm;
+	};
+
+	sound-ak5558 {
+		fsl,tdm;
+	};
+};
-- 
1.7.9.5

